domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -97,15 +336,62 @@ class ProfileManagementController @Inject constructor(
     return ProfileId.newBuilder().setInternalId(currentProfileId).build()
   }
 
-  /**
-   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
-   * Checks to ensure that profileId corresponds to an existing profile.
-   *
-   * @param profileId the ID corresponding to the profile being set.
-   * @return a [LiveData] that indicates the success/failure of this set operation.
-   */
-  fun setCurrentProfileId(profileId: ProfileId): LiveData<AsyncResult<Any?>> {
-    currentProfileId = profileId.internalId
-    return MutableLiveData(AsyncResult.success<Any?>(null))
+  private fun isUniqueName(newName: String, profileDatabase: ProfileDatabase): Boolean {
+    val lowerCaseNewName = newName.toLowerCase()
+    profileDatabase.profilesMap.values.forEach {
+      if (it.name.toLowerCase(Locale.getDefault()) == lowerCaseNewName) {
+        return false
+      }
+    }
+    return true
+  }
+
+  private fun saveImageToInternalStorage(avatarImagePath: Uri, profileDir: File): String? {
+    val bitmap = MediaStore.Images.Media.getBitmap(context.contentResolver, avatarImagePath)
+    val imageFile = File(profileDir, "profile.png")
```

Comment:
Suggest storing the file name in a constant so that it can be easily shared.

Also, perhaps name this ``profile_avatar.png`` to be more specific about what the image contains.